QoL/Modules/PatchedBosses.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Linq;
+using HutongGames.PlayMaker;
+using MonoMod.Cil;
+using UnityEngine;
+using Vasi;
+
+namespace QoL.Modules
+{
+    public class PatchedBosses : FauxMod
+    {
+        [SerializeToSetting]
+        public static bool SporeShroomHollowKnight = true;
+
+        [SerializeToSetting]
+        public static bool NoBackrolls = false;
+
+        [SerializeToSetting]
+        public static bool SleepyCrystalGuardian = false;
+
+
+        public override void Initialize()
+        {
+            IL.ExtraDamageable.RecieveExtraDamage += AllowRecieveExtraDamage;
+            On.DamageEffectTicker.OnTriggerExit2D += KeepDamagingInactives;
+
+            On.PlayMakerFSM.OnEnable += BossFsmChanges;
+        }
+
+        public override void Unload()
+        {
+            IL.ExtraDamageable.RecieveExtraDamage -= AllowRecieveExtraDamage;
+            On.DamageEffectTicker.OnTriggerExit2D -= KeepDamagingInactives;
+
+            On.PlayMakerFSM.OnEnable -= BossFsmChanges;
+        }
+
+        private void BossFsmChanges(On.PlayMakerFSM.orig_OnEnable orig, PlayMakerFSM self)
+        {
+            orig(self);
+            if (self.FsmName == "Black Knight" && NoBackrolls)
```

Comment:
Are these not available after a scene change? It's fine if they aren't but this seems heavy-handed for a fairly simple change